rest/diagnostic_api.go

```diff
@@ -0,0 +1,222 @@
+//  Copyright 2013-Present Couchbase, Inc.
+//
+//  Use of this software is governed by the Business Source License included
+//  in the file licenses/BSL-Couchbase.txt.  As of the Change Date specified
+//  in that file, in accordance with the Business Source License, use of this
+//  software will be governed by the Apache License, Version 2.0, included in
+//  the file licenses/APL2.txt.
+
+package rest
+
+import (
+	"fmt"
+
+	"github.com/couchbase/sync_gateway/auth"
+	"github.com/couchbase/sync_gateway/base"
+	"github.com/couchbase/sync_gateway/channels"
+	"github.com/gorilla/mux"
+)
+
+// each field in this response is a map of collection names to a map of channel names to grant history info
+type getAllChannelsResponse struct {
+	AdminGrants       map[string]map[string]auth.GrantHistory            `json:"admin_grants,omitempty"`
```

Comment:
> Adding this field means that it is potentially going to change existing APIs

Me and Ben had long discussions when the solution was designed, and this was taken into account. If source is marked as omitempty I dont think its gonna break any existing APIs. Since thats the case, why would I need another struct for grant histories?

> Source is not always set.

We also had a discussion on whether the response from this endpoint should still contain source and the conclusion was that we should only keep it for role grants